shopinvader/services/cart.py

```diff
@@ -226,8 +226,16 @@ def _sale_order_line_onchange(self, vals):
         # - fiscal_position_id
         # - pricelist_id
         so_line_obj = self.env["sale.order.line"].suspend_security()
+        # Apply play_onchanges without changing the env (not in draft mode)
+        new_values = so_line_obj.play_onchanges(vals, vals.keys())
         with self._ensure_ctx_lang(vals):
-            new_values = so_line_obj.play_onchanges(vals, vals.keys())
+            # The play_onchanges in draft mode is not supported (to update
+            # every fields). So apply it and just update the 'name' if filled.
+            new_values_lang = so_line_obj.play_onchanges(vals, vals.keys())
```

Comment:
We must have a unittest to reproduce the problem. This changes has a high performance impact. Therefore we must find the root cause.